x-pack/plugins/observability_solution/slo/server/services/reset_slo.ts

```diff
@@ -113,6 +117,17 @@ export class ResetSLO {
     return resetSLOResponseSchema.encode(updatedSlo);
   }
 
+  private async assertExpectedIndicatorSourceIndexPrivileges(slo: SLODefinition) {
```

Comment:
Seems like this can be moved to a helper as this is repeated in every class. 

Alternatively could create a parent class which implements any generic functions for create/update/reset slo classes

Reply:
resolved with https://github.com/elastic/kibana/pull/199233/commits/34f3cdb47f079a7bd6f40eb539a3cddcaace18ed